### ISSUE TITLE
The porting status of this project

### ISSUE THREAD

**Issue by Peter9606** [bponsler/message_filters#1]
Hi, bponsler,

Glad to see you are working on this project which is quite critical to me. I want to ask what's the porting status, is it ready to use or only with build tool changed to ament?
If there are extra efforts to put on message_filters, and you are okay with it, I'd like also contribute with this.

thanks,
Peter Han

**Comment by bponsler** [comment]
@Peter9606 -- it's in progress. Currently it is only at a state where it compiles and installs (so it can be linked against other packages). I'm in the process of fixing/porting its functionality, but would welcome any contributions you'd like to make!

**Comment by bponsler** [comment]
@Peter9606 -- I just pushed commit b60fc8cd992676f2aaeda1a6e25fe58092b7bce3 which improves support for some pieces of message_filters. There's still a lot that is not supported or working correctly.

**Comment by Peter9606** [comment]
Is there any porting guide you are following ? Can you share it with me?
In case you are not noticed this thread, @dirk-thomas talked a lot useful things on reuse ROS1 in ROS2 [here](https://discourse.ros.org/t/software-reuse-in-ros1-and-ros2/2262/5)

**Comment by bponsler** [comment]
Mostly just following [ROS 2 the migration guide](https://github.com/ros2/ros2/wiki/Migration-Guide) and searching through the ROS 2 source code when it doesn't cover specific cases. Yes, I did see that thread and it does have a lot of useful information

**Comment by dirk-thomas** [comment]
Is there a reason why you didn't fork the repo with the ROS 1 code? It would make it easier to compare the changes.

**Comment by bponsler** [comment]
@dirk-thomas -- great question...I originally felt it was unnecessary to fork all of ros_comm just to port message_filters. Especially since most of ros_comm probably won't be ported to ROS 2.

Anyway, you make a very valid point...I just forked ros_comm from lunar-devel and added my changes there so it is easier to track (see new location of [message_filters](https://github.com/bponsler/ros_comm/tree/ros2-devel/utilities/message_filters). I'll update the readme for this package, and other links to direct people to the new location and eventually delete this repo in favor of the fork. Thanks for pointing this out.

**Comment by Peter9606** [comment]
@bponsler Do you have any plan such as by when this porting should be done?

**Comment by bponsler** [comment]
No specific plans, just porting functionality as I need it. message_filters isn't specifically on my critical path (yet).

**Comment by Peter9606** [comment]
Thanks for letting me know you have no plan yet :).  I'll contribute as many efforts as I can, as my current project depends on it. :(

**Comment by bponsler** [comment]
Sounds good, I'll try to help you out as I can. Let's make sure we work off of https://github.com/bponsler/ros_comm/tree/ros2-devel/utilities/message_filters instead though so it's easier to track changes.

**Comment by Peter9606** [comment]
Absolutely. I'm currently working on a mis-dependent error, hope will PR to https://github.com/bponsler/ros_comm/tree/ros2-devel/utilities/message_filters project in next few days.
